core/src/main/java/org/infinispan/marshall/protostream/impl/SerializationContextRegistry.java

```diff
@@ -0,0 +1,33 @@
+package org.infinispan.marshall.protostream.impl;
+
+import org.infinispan.protostream.BaseMarshaller;
+import org.infinispan.protostream.FileDescriptorSource;
+import org.infinispan.protostream.SerializationContext;
+import org.infinispan.protostream.SerializationContextInitializer;
+import org.infinispan.protostream.config.Configuration;
+
+/**
+ * Manages {@link SerializationContext} across modules for use by various components.
+ *
+ * @author Ryan Emerson
+ * @since 10.0
+ */
+public interface SerializationContextRegistry {
+
+   void addProtoFile(MarshallerType type, FileDescriptorSource fileDescriptorSource);
+
+   void addMarshaller(MarshallerType type, BaseMarshaller marshaller);
+
+   void addContextInitializer(MarshallerType type, SerializationContextInitializer sci);
+
+   void addConfiguration(MarshallerType type, Configuration config);
```

Comment:
When would I use this method? This is the first time I have ever seen this `Configuration` clas.

Reply:
It's required by remote-query, as it defines a bunch of custom annotations https://github.com/infinispan/infinispan/blob/03895a50132fbf312dc05d456c8a59e278f965a5/remote-query/remote-query-server/src/main/java/org/infinispan/query/remote/impl/indexing/IndexingMetadata.java#L292-L343